app/section/sections/mobile-operating-systems/ui.test.js

```diff
@@ -0,0 +1,192 @@
+  const otherCheckbox = operatingSystemsQuestion.find('.nhsuk-checkboxes__item:nth-child(3)');
+
+  await t
+    .expect(appleIOSCheckbox.find('input:checked').exists).ok()
+    .expect(androidCheckbox.find('input:checked').exists).ok()
+    .expect(otherCheckbox.find('input:checked').exists).ok()
+});
+
+test('should render the operating systems description question', async (t) => {
+  const question = 'operating-systems-description';
+  await pageSetup(t);
+  const operatingSystemsDescriptionQuestion = Selector(`[data-test-id="question-${question}"]`);
+  const expectedDescription = sectionManifest.questions[question];
+  await t
+    .expect(operatingSystemsDescriptionQuestion.find('label.nhsuk-label').innerText).eql(expectedDescription.mainAdvice)
+    .expect(operatingSystemsDescriptionQuestion.find('span.nhsuk-hint').innerText).eql(expectedDescription.additionalAdvice)
+    .expect(operatingSystemsDescriptionQuestion.find('textarea').count).eql(1)
+    .expect(operatingSystemsDescriptionQuestion.find('[data-test-id="textarea-field-footer"]').innerText).eql(expectedDescription.footerAdvice);
+});
+
+test('should populate the operating systems description textarea with existing data', async (t) => {
+  await pageSetup(t, 200, mobileOperatingSystemsMarketingData);
+  const descriptionQuestion = Selector('[data-test-id="question-operating-systems-description"]');
+
+  await t
+    .expect(descriptionQuestion.find('textarea').value).eql(mobileOperatingSystemsMarketingData['operating-systems-description']);
+});
+
+test('should show error summary and validation for operating systems description question when it exceeds the maxLength', async (t) => {
+  const question = 'operating-systems-description';
+  await pageSetup(t);
+  nock('http://localhost:8080')
+    .put(sectionApiUrl)
+    .reply(400, {
+      maxLength: [question],
+    });
+
+  const errorSummary = Selector('[data-test-id="error-summary"]');
+  const errorSummaryList = Selector('.nhsuk-error-summary__list');
+  const operatingSystemsDescription = Selector(`[data-test-id="question-${question}"]`);
+
+  const submitButton = Selector('[data-test-id="section-submit-button"]');
+  const expectedDescription = sectionManifest.questions[question];
+  await t
+    .expect(errorSummary.exists).notOk()
+    .typeText(operatingSystemsDescription, `${'x'.repeat(501)}`, { paste: true })
+    .click(submitButton.find('button'))
+    .expect(errorSummary.exists).ok()
+    .expect(errorSummaryList.find('li').count).eql(1)
+    .expect(errorSummaryList.find('li:nth-child(1)').innerText).eql(expectedDescription.errorResponse.maxLength)
+    .expect(errorSummaryList.find('li:nth-child(1) a').getAttribute('href')).eql('#operating-systems-description')
+    .expect(operatingSystemsDescription.find('.nhsuk-textarea--error').exists).ok()
+    .expect(operatingSystemsDescription.find('.nhsuk-error-message').innerText).eql(`Error:\n${expectedDescription.errorResponse.maxLength}`);
+});
+
```

Comment:
Could do with a test to validate the details of the operation systems checkbox 'required' error message

Reply:
added